app/src/main/java/com/testfairy/TestFairyFeedbackOverlay.java

```diff
@@ -0,0 +1,614 @@
+						}
+					});
+				} else {
+					activity.runOnUiThread(new Runnable() {
+						@Override
+						public void run() {
+							instance.removeOverlay(false);
+						}
+					});
+					Log.e(TAG, "There is no view to attach TestFairy feedback overlay");
+				}
+			}
+		} else if (!(activity instanceof ProvideFeedbackActivity)) {
+			instance = new TestFairyFeedbackOverlay();
+			final View root = activity.getWindow().getDecorView().findViewById(android.R.id.content);
+
+			if (root != null && root instanceof FrameLayout) {
+				activity.runOnUiThread(new Runnable() {
+					@Override
+					public void run() {
+						visible = visibleBeforePause;
+						instance.attachOverlay((FrameLayout) root);
+					}
+				});
+			} else {
+				Log.e(TAG, "There is no view to attach TestFairy feedback overlay");
+			}
+		}
+	}
+
+	static private void onPause(Activity activity) {
+		Log.d(TAG, "TestFairyFeedbackOverlay is detaching from " + activity.getClass().getSimpleName());
+
+		if (instance != null) {
+			synchronized (instance) {
+				activity.runOnUiThread(new Runnable() {
+					@Override
+					public void run() {
+						visibleBeforePause = visible;
+						instance.removeOverlay(false);
+						instance = null;
+					}
+				});
+			}
+		}
+	}
+
+	static private float dpToPx(Context context, float dp) {
+		Resources r = context.getResources();
+		return TypedValue.applyDimension(
+				TypedValue.COMPLEX_UNIT_DIP,
+				dp,
+				r.getDisplayMetrics()
+		);
+	}
+
+	/**
+	 * shows a bitmap as if it had rounded corners. based on :
+	 * http://rahulswackyworld.blogspot.co.il/2013/04/android-drawables-with-rounded_7.html
+	 * easy alternative from support library: RoundedBitmapDrawableFactory.create( ...) ;
+	 */
+	static private class RoundedCornersDrawable extends BitmapDrawable {
+		private final BitmapShader bitmapShader;
+		private final Paint p;
+		private final RectF rect;
+		private final float borderRadius;
+		private final boolean enableBackground;
+
+		public RoundedCornersDrawable(final Resources resources, final Bitmap bitmap, final float borderRadius, boolean enableBackground) {
+			super(resources, bitmap);
+
+			this.bitmapShader = new BitmapShader(getBitmap(), Shader.TileMode.CLAMP, Shader.TileMode.CLAMP);
+
+			Bitmap b = getBitmap();
+
+			this.p = getPaint();
+			this.p.setAntiAlias(true);
+			this.p.setShader(bitmapShader);
+
+			int w = b.getWidth(), h = b.getHeight();
+			this.rect = new RectF(0, 0, w, h);
+
+			this.borderRadius = borderRadius < 0 ? 0.15f * Math.min(w, h) : borderRadius;
+			this.enableBackground = enableBackground;
+		}
+
+		@Override
+		public void draw(final Canvas canvas) {
+			if (enableBackground) {
+				Paint paint = new Paint();
+				paint.setColor(ICON_BG_COLOR);
+				canvas.drawRoundRect(rect, borderRadius, borderRadius, paint);
+			}
+
+			canvas.drawRoundRect(rect, borderRadius, borderRadius, p);
+		}
+	}
+
+	static private class RoundedImageView extends ImageView {
```

Comment:
Could we just have the png pre-rendered with a background color and a transparent round circle? :))

Reply:
This is what happens when a programmer designs a UI, still better than Tesla truck. :) Imma draw something with Krita, let me know how it looks in the next commit.